### PR TITLE
Add zsh completion for docker log options 'max-buffer-size|mode'

### PR DESCRIPTION
**Pull request by sdurrheimer**
ref. #28762

This can be planned for `1.14.0`.

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -261,8 +262,12 @@ __docker_complete_log_options() {
     if compset -P '*='; then
         case "${${words[-1]%=*}#*=}" in
             (syslog-format)
-                syslog_format_opts=('rfc3164' 'rfc5424' 'rfc5424micro')
-                _describe -t syslog-format-opts "Syslog format Options" syslog_format_opts && ret=0
+                local opts=('rfc3164' 'rfc5424' 'rfc5424micro')
+                _describe -t syslog-format-opts "syslog format options" opts && ret=0
+                ;;
+            (mode)
+                local opts=('blocking' 'nonblocking')
```
Comment (cpuguy83) [inline contrib/completion/zsh/_docker]: I believe this is currently `non-blocking`, perhaps it should not be hyphenated though?

Comment (sdurrheimer) [inline contrib/completion/zsh/_docker]: Hum sorry, I based my PR on @albers bash's completion (https://github.com/docker/docker/blob/master/contrib/completion/bash/docker#L793) but looks like it is indeed `non-blocking` (https://github.com/docker/docker/blob/master/api/types/container/host_config.go#L234)

Comment (albers) [inline contrib/completion/zsh/_docker]: O yes, I was wrong:
```bash
$ docker run --log-opt mode=nonblocking busybox echo hello
docker: Error response from daemon: logger: logging mode not supported: nonblocking.
See 'docker run --help'.
$ docker run --log-opt mode=non-blocking busybox echo hello
hello
```

**Comment by sdurrheimer** [comment]
PR fixed.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**